lib/Common/Memory/Recycler.h

```diff
@@ -155,6 +155,46 @@ template<ObjectInfoBits infoBits>
 struct InfoBitsWrapper{};
 
 
+#if ENABLE_WEAK_REFERENCE_REGIONS
+template<typename T>
+static constexpr bool is_pointer = false;
+template<typename K>
+static constexpr bool is_pointer<K*> = true;
+
+template<typename T>
+class RecyclerWeakReferenceRegionItem {
+    static_assert(is_pointer<T>, "Weak references must be to pointer types");
+    friend class Recycler;
+public:
+    RecyclerWeakReferenceRegionItem() : ptr(T()), heapBlock(nullptr) {};
+    operator T() const { return ptr; }; // Mask off the low bit, used for concurrent marking
```

Comment:
remove comment (and maybe add a new comment on heapBlock saying it gets the flag)

Reply:
Good catch, updated.